src/lib/db/context-field-store.ts

```diff
@@ -68,18 +83,48 @@ class ContextFieldStore implements IContextFieldStore {
     }
 
     async getAll(): Promise<IContextField[]> {
-        const rows = await this.db
-            .select(COLUMNS)
-            .from(TABLE)
-            .orderBy('name', 'asc');
-
-        return rows.map(mapRow);
+        if (this.flagResolver.isEnabled('segmentContextFieldUsage')) {
+            const rows = await this.db
```

Comment:
The query looks scary, but tested and query plan is reasonable. Furthermore since the base query is getting **context fields** and there are not so many of them, it is not a issue.

Reply:
what was the biggest data set that you run it on? E.g. have you checked context fields used in 5k features?

Reply:
Yes, 100-200ms. Lighting fast.